File: EventToolsCore/ClientApp/src/data/SearchResponse.ts
import { observable } from 'mobx';

import { Entity } from './Entity';

export class FacetCount {
    public category: string;

    public value: string;

    public count: number;
}

export class SearchResponse {
    public entities: Entity[];

    public facets: FacetCount[];

    public totalCount: number;

    public currentPage: number;

    public previousPageToken: string;

    public pextPageToken: string;
}

File: EventToolsCore.Domain/ClientDtos/SearchResponse.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace EventToolsCore.Domain.ClientDtos
{
    public class SearchResponse
    {
        public EntityBase[] Entities;

        public FacetCount[] Facets;

        public int TotalCount;

        public int CurrentPage;

        public string PreviousPageToken;

        public string NextPageToken;
    }
}
